packages/gatsby-plugin-image/src/components/picture.tsx

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable filenames/match-regex */
+import React, {
+  FunctionComponent,
+  ImgHTMLAttributes,
+  forwardRef,
+  LegacyRef,
+} from "react"
+import * as PropTypes from "prop-types"
+
+export interface IResponsiveImageProps {
+  sizes?: string
+  srcSet: string
+}
+
+export type SourceProps = IResponsiveImageProps &
+  (
+    | {
+        media: string
+        type?: string
+      }
+    | {
+        media?: string
+        type: string
+      }
+  )
+
+type FallbackProps = { src: string } & Partial<IResponsiveImageProps>
+
+type ImageProps = ImgHTMLAttributes<HTMLImageElement> & {
+  src: string
+  alt: string
+  shouldLoad: boolean
+  innerRef: LegacyRef<HTMLImageElement>
+}
+
+export type PictureProps = ImgHTMLAttributes<HTMLImageElement> & {
+  fallback?: FallbackProps
+  sources?: Array<SourceProps>
+  alt: string
+  shouldLoad?: boolean
+  forceWrapper?: boolean
+}
+
+const Image: FunctionComponent<ImageProps> = function Image({
+  src,
+  srcSet,
+  loading,
+  alt = ``,
+  shouldLoad,
+  innerRef,
+  ...props
+}) {
+  return (
+    <img
+      {...props}
+      decoding="async"
+      loading={loading}
+      src={shouldLoad ? src : undefined}
+      data-src={!shouldLoad ? src : undefined}
+      srcSet={shouldLoad ? srcSet : undefined}
+      data-srcset={!shouldLoad ? srcSet : undefined}
+      alt={alt}
+      ref={innerRef}
+    />
+  )
+}
+
+export const Picture = forwardRef<HTMLImageElement, PictureProps>(
+  function Picture(
+    { fallback, sources = [], shouldLoad = true, forceWrapper, ...props },
+    ref
+  ) {
+    const fallbackImage = (
+      <Image {...props} {...fallback} shouldLoad={shouldLoad} innerRef={ref} />
+    )
+
+    if (!sources.length && !forceWrapper) {
+      return fallbackImage
+    }
```

Comment:
Can we remove forcewrapper?
```suggestion
    if (!sources.length) {
      return fallbackImage
    }
```